### ISSUE TITLE
Issues with Resets and Memory Leak in Default Training

### ISSUE THREAD

**Issue by isaac-racine** [real-stanford/umi-on-legs#8]
Hello,

I'm testing the default training configuration "combo_go2ARX5_pickle_reaching_extreme" and ran into some issues that I could use help with.

**Expected Training Outcome:** Without modifying the code, should the robot be able to follow the tossing end-effector (EE) trajectory? For me at around the 200 iteration mark, the robots start resetting instantaneously, seemingly due to a termination criterion. This behavior continues until the end of the full 20,000 iterations. So the result are not good.

**GPU Memory Leak:** Also starting at around the 200 iteration mark, GPU memory usage steadily increases over several hundred iterations until the training crashes. I made a modification in env.py to address this:

Original:

`     self.obs_history = torch.cat((self.obs_history[:, 1:, :], obs.unsqueeze(1)), dim=1)`

Modified:

```
    self.obs_history[:, :-1, :] = self.obs_history[:, 1:, :]
    self.obs_history[:, -1, :] = obs
```

This change seems to prevent the memory increase, but training results remain the same. Do you have any insights on this issue?

System Specs:

    OS: Ubuntu 22.04
    GPU: NVIDIA GeForce RTX 4090 (25 GB memory)
    Environment: Miniconda3 with IsaacGym_Preview_4

Thank you very much for your help!

**Comment by huy-ha** [comment]
I don't expect the instant resets. Just to clarify, you are using the exact code version on master right now with no modifications, the configuration in `combo_go2ARX5_pickle_reaching_extreme` without overriding any hyperparameters other than the ones included in the default command, and using our task trajectory dataset?

For clarity, this is the default command with default overrides I provided in the README
```
python scripts/train.py env.sim_device=cuda:0 env.graphics_device_id=0 env.tasks.reaching.sequence_sampler.file_path=data/tossing.pkl
```

For the GPU memory leak, I've observed Isaac Gym leaks memory due to contacts. It can happen much much later in training (say 20k iterations), but 200 is way too early. 

I have trained on a very similar if not identical system setup before, so I don't believe it's a systems issue.

**Comment by isaac-racine** [comment]
Thank you for the fast reply! 

Yes, I am using the current code on the main branch with no modifications, with your tossing.pkl dataset and using the default command that you showed. I am gonna try it on a different pc with a new ubuntu 22.04 and let you know if it works properly.

**Comment by isaac-racine** [comment]
**Update**

So I tried to run the training on a new pc and the results are the same. Without changing the code the gpu memory starts to increase and the training crashes a little bit after 500 iterations. The training seems to go well before that so it must be the contacts.

**Comment by yolo01826** [comment]
![image](https://github.com/user-attachments/assets/ad37b5ac-f124-4b30-951b-852a61a7b17b)
I encountered the same issue in the cup placement task without modifying the code.
（ubuntu20.04，RTX4090）

**Comment by huy-ha** [comment]
Ah! When using cup in the wild trajectories, you should add some z height to the trajectory. For instance, adding 'env.tasks.reaching.sequence_sampler. add_random_height_range=[0.4,0.5]' to the command should randomly add 40cm to 50cm to the trajectory z index, which is more realistic for this particular task. This would also prevent the robot from crawling on the ground the entire time, which should prevent contact buffers from taking up so much memory.

**Comment by yolo01826** [comment]
thanks a lot, it works🌹

**Comment by hhhappyshow** [comment]
Hi, isaac-racine,

Thank you for you kind sharing of your experience!

I've encountered similar GPU Memory Leak problem as yours, and I tried your method to modify the code.

> 
> Original:
> 
> ` self.obs_history = torch.cat((self.obs_history[:, 1:, :], obs.unsqueeze(1)), dim=1)`
> 
> Modified:
> 
> ```
>     self.obs_history[:, :-1, :] = self.obs_history[:, 1:, :]
>     self.obs_history[:, -1, :] = obs
> ```
> 
However, I found that the training result is really poor after the modification. I think it's because that when the torch was copying the data, it dosen't follow the order which makes the data of the new obs_history is wrong. So I tried the new modification: 

> New modification:
> ```
>     self.obs_history = torch.roll(self.obs_history, shifts=-1, dims=1)
>     self.obs_history[:, -1, :] = obs
and then this time,the training result is the same as the original one .
Did you have this experience as well?
